erc20-wrapper-registry/test/compliantFactory.ts

```diff
@@ -0,0 +1,70 @@
+import { expect } from "chai";
+import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
+import type { Signers } from "./types";
+
+// import { CompliantFactory, CompliantFactory__factory } from "../typechain-types";
+import { ethers } from "hardhat";
+import { SignerWithAddress } from "@nomicfoundation/hardhat-ethers/signers";
+import { deployCompliantFactoryFixture } from "./compliantFactory.fixture";
+import { compliantERC20Bytecode } from "../frontend/abis/VerifiedAirdrop";
+import { CompliantERC20__factory, CompliantFactory } from "../typechain-types";
+import { id } from "ethers";
+
+
+describe("CompliantFactory Unit tests", function () {
+  before(async function () {
+    this.signers = {} as Signers;
+
+    const signers: SignerWithAddress[] = await ethers.getSigners();
+    this.signers.owner = signers[0];
+    this.signers.admin = signers[1];
+    this.signers.user = signers[2];
+
+    this.loadFixture = loadFixture;
+  });
+
+  describe("CompliantFactory", function () {
+    beforeEach(async function () {
+      const { compliantFactory, mockERC20 } = await this.loadFixture(deployCompliantFactoryFixture);
+      this.compliantFactory = compliantFactory;
+      this.mockERC20 = mockERC20;
+    });
+
+    shouldBehaveLikeCompliantFactory();
+  });
+});
+
+
+export function shouldBehaveLikeCompliantFactory(): void {
+  describe("create2", async function () {
```

Comment:
I left it in case we want to reintroduce the create2, as tests are not run on the pipeline, would make more sense to keep the uncommented tho, otherwise we should also comment the mocks and fixture, but this would look quite bad